backend/src/test/java/com/google/sps/FindMatchQueryTest.java

```diff
@@ -0,0 +1,149 @@
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match match = query.findMatch(Arrays.asList(participantA), participantB);
+    
+    assertThat(Objects.isNull(match)).isTrue();
+  }
+
+  @Test
+  public void threeParticipants13() {
+    // Three participants, 1st and 2nd aren't compatible, but 1st and 3rd are
+    Participant participantA = new Participant(ID_DEFAULT, PERSON_A, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0250PM, TIMEZONE_DEFAULT, DURATION_45_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantC = new Participant(ID_DEFAULT, PERSON_C, TIME_0600PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match match = query.findMatch(Arrays.asList(participantA, participantB), participantC);
+    
+    assertThat(match.getFirstParticipant().getLdap().equals(PERSON_C));
+    assertThat(match.getSecondParticipant().getLdap().equals(PERSON_A));
+    assertThat(match.getDuration() == DURATION_60_MINUTES);
+  }
+
+  @Test
+  public void threeParticipants23() {
+    // Three participants, 1st and 2nd aren't compatible, but 2nd and 3rd are
+    Participant participantA = new Participant(ID_DEFAULT, PERSON_A, TIME_0250PM, TIMEZONE_DEFAULT, DURATION_30_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantC = new Participant(ID_DEFAULT, PERSON_C, TIME_0600PM, TIMEZONE_DEFAULT, DURATION_60_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match match = query.findMatch(Arrays.asList(participantA, participantB), participantC);
+    
+    assertThat(match.getFirstParticipant().getLdap().equals(PERSON_C));
+    assertThat(match.getSecondParticipant().getLdap().equals(PERSON_B));
+    assertThat(match.getDuration() == DURATION_60_MINUTES);
+  }
+
+  @Test
+  public void threeParticipantsTwoMatches() {
+    // Three participants, 1st & 2nd, 1st & 3rd are compatible but only return 1st & 3rd
+    Participant participantA = new Participant(ID_DEFAULT, PERSON_A, TIME_0400PM, TIMEZONE_DEFAULT, DURATION_30_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB = new Participant(ID_DEFAULT, PERSON_B, TIME_0600PM, TIMEZONE_DEFAULT, DURATION_15_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantC = new Participant(ID_DEFAULT, PERSON_C, TIME_0800PM, TIMEZONE_DEFAULT, DURATION_45_MINUTES, TIMESTAMP_DEFAULT);
+    
+    Match match = query.findMatch(Arrays.asList(participantA, participantB), participantC);
+    
+    assertThat(match.getFirstParticipant().getLdap().equals(PERSON_C));
+    assertThat(match.getSecondParticipant().getLdap().equals(PERSON_A));
+    assertThat(match.getDuration() == DURATION_60_MINUTES);
```

Comment:
assertThat(match.getDuration()).isEqualTo(DURATION_60_MINUTES) otherwise this is an incomplete assertion